hazelcast/src/main/java/com/hazelcast/map/impl/MapSplitBrainHandlerService.java

```diff
@@ -58,14 +58,14 @@ protected Runnable newMergeRunnable(Collection<RecordStore> mergingStores) {
     protected void onStoreCollection(RecordStore recordStore) {
         assertRunningOnPartitionThread();
 
-        ((DefaultRecordStore) recordStore).clearIndexedData(false);
+        ((DefaultRecordStore) recordStore).clearOtherResourcesThanStorage(true);
```

Comment:
Now in addition to clearing indexes, we do this:
```
clearMapStore();
        clearLockStore();
        clearIndexedData(true);
```
Here we clear the lock store even though it might not yet have been collected by the `LockService`. I'm not sure if this is ok.

Reply:
I've double-checked and yes, we need to avoid removing data here that will be collected by a different service. Map store and indexes are fine but lock store should be removed as we will probably remove locks before they are collected by the LockService.

Reply:
removed `clearLockStore`.